src/viewer/NoViewer.jsx

```diff
@@ -1,26 +1,55 @@
+/* global cozy */
 import React from 'react'
 import classNames from 'classnames'
 import { downloadFile } from 'cozy-client'
 import { isCordova } from 'drive/mobile/lib/device'
-import { openOfflineFile, openOnlineFile } from 'drive/mobile/lib/filesystem'
+import {
+  openOfflineFile,
+  temporarySave,
+  openFileWithCordova
+} from 'drive/mobile/lib/filesystem'
 import { translate } from 'cozy-ui/react/I18n'
 import Button from 'cozy-ui/react/Button'
 
 import styles from './styles'
 
-const OpenWithCordova = ({ t, file }) => (
-  <Button
-    theme="regular"
-    className={styles['pho-viewer-noviewer-download']}
-    onClick={() =>
-      file.isAvailableOffline ? openOfflineFile(file) : openOnlineFile(file)
-    }
-  >
-    {t('Viewer.noviewer.openWith')}
-  </Button>
-)
+class OpenWithCordovaButton extends React.Component {
+  state = {
+    loading: false
+  }
+
+  onClick = () => {
+    const { file, onError } = this.props
+    file.isAvailableOffline
+      ? openOfflineFile(file)
+      : (async () => {
+          this.setState(state => ({ ...state, loading: true }))
+          const response = await cozy.client.files.downloadById(file.id)
+          const blob = await response.blob()
+          const localFile = await temporarySave(blob, file.name)
+          this.setState(state => ({ ...state, loading: false }))
+          return openFileWithCordova(localFile.nativeURL, file.mime).catch(
+            onError
```

Comment:
Maybe the `onError` should be the result of a `catch` block that also wraps the download and saving parts, just in case.